ethereum/trie/src/main/java/tech/pegasys/pantheon/ethereum/trie/Proof.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.trie;
+
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.util.List;
+import java.util.Optional;
+
+public class Proof<V> {
+  private final V value;
+
+  private final List<BytesValue> proofRelatedNodes;
+
+  public Proof(final V value, final List<BytesValue> proofRelatedNodes) {
```

Comment:
(nit) If you accept an `Optional<V>` in the constructor and make `value` an `Optional<V>`, you can collapse the two constructors into one.  In our codebase we tend to keep values wrapped in `Optional`'s internally rather than holding onto potentially `null` values.